src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -733,80 +736,56 @@ private void GenerateResourceLoader(IndentedStringBuilder writer)
 
 			writer.AppendLineIndented($"global::Windows.ApplicationModel.Resources.ResourceLoader.AddLookupAssembly(GetType().Assembly);");
 
-			foreach (var reference in _metadataHelper.Compilation.ExternalReferences)
+			foreach (var reference in _metadataHelper.Compilation.References)
 			{
-				string? GetFilePath()
+                if(_metadataHelper.Compilation.GetAssemblyOrModuleSymbol(reference) is IAssemblySymbol assembly)
+                {
+					BuildResourceLoaderFromAssembly(writer, assembly);
+                }
+                else
+                {
+                    throw new InvalidOperationException($"Unsupported resource type for {reference.Display} ({reference.GetType()})");
+                }
+            }
+		}
+
+        private void BuildResourceLoaderFromAssembly(IndentedStringBuilder writer, IAssemblySymbol assembly)
+        {
+			if (assembly.GetAllAttributes().Any(a =>
```

Comment:
This should be sufficient. `GetAllAttributes` can only behave differently for `INamedTypeSymbol`s

```suggestion
			if (assembly.GetAttributes().Any(a =>
```